core/components/organisms/textField/TextFieldWithInput.tsx

```diff
@@ -0,0 +1,239 @@
+  /**
+   * Handler to be called when `Input` gets focus
+   */
+  onFocus?: (e: React.FocusEvent<HTMLInputElement>) => void;
+  /**
+   * Handler to be called when paste event occur inside `Input`
+   */
+  onPaste?: React.ClipboardEventHandler<HTMLInputElement>;
+  /**
+   * Custom Icon Component to be passed to Input to replace Clear Icon in the right
+   */
+  actionIcon?: React.ReactElement<IconProps>;
+  /**
+   * Add text below `input`
+   */
+  helpText?: string;
+}
+
+export const TextFieldWithInput = (props: TextFieldInputProps) => {
+  const {
+    label,
+    size,
+    minWidth,
+    defaultValue,
+    name,
+    placeholder,
+    value,
+    icon,
+    inlineLabel,
+    required,
+    error,
+    info,
+    onChange,
+    onClick,
+    onClear,
+    onBlur,
+    onFocus,
+    onPaste,
+    className,
+    actionIcon,
+    autoFocus,
+    disabled,
+    readOnly,
+    autoComplete,
+    minLength,
+    maxLength,
+    max = 200,
+    pattern,
+    helpText = ' ',
+  } = props;
+
+  const [inputText, setInputText] = React.useState<string>('');
+
+  const onChangeHandler = (e: React.ChangeEvent<HTMLInputElement>) => {
+    setInputText(e.target.value);
+    if (onChange) onChange(e);
+  };
+
+  const inputError = error || inputText.length > max;
+
+  return (
+    <div>
+      {label && (
+        <Label required={required} withInput={true}>
+          {label}
+        </Label>
+      )}
+      <Input
+        size={size}
+        minWidth={minWidth}
+        defaultValue={defaultValue}
+        name={name}
+        placeholder={placeholder}
+        value={value}
+        icon={icon}
+        inlineLabel={inlineLabel}
+        error={inputError}
+        info={info}
+        onChange={onChangeHandler}
+        onClick={onClick}
+        onClear={onClear}
+        onBlur={onBlur}
+        onFocus={onFocus}
+        onPaste={onPaste}
+        actionIcon={actionIcon}
+        autoFocus={autoFocus}
+        disabled={disabled}
+        readOnly={readOnly}
+        autoComplete={autoComplete}
+        className={className}
+        minLength={minLength}
+        maxLength={maxLength}
+        pattern={pattern}
+      />
+      <div className="d-flex justify-content-between" style={{ minWidth }}>
+        <HelpText
+          message={helpText ? helpText : ' '}
+          className="d-flex"
+          error={helpText.trim().length > 0 ? inputError : undefined}
+        />
+        <div className="mt-3 d-flex">
+          <Text
+            appearance="subtle"
+            className="pr-2"
+            color={inputText.length > max ? 'alert' : undefined}
+            size="small"
+            weight="medium"
+          >
+            {inputText.length}
+          </Text>
+          <Text appearance="subtle" className="pr-2" size="small" weight="medium">
+            /
+          </Text>
+          <Text appearance="subtle" size="small" weight="medium">
+            {max}
+          </Text>
+        </div>
+      </div>
+    </div>
```

Comment:
Please create a separate component for this as same code is used in both Textarea and Input